### DISCUSSION TITLE
Add a sqlite format

### DISCUSSION THREAD

**Discussion by badele** [koordinates/kart#526] [Ideas]
Hi, actually the Kart project support GeoPackage, PostGIS, Microsoft SQL Server and MySQL format.

Have you scheduled add the sqlite format ?

**Note**: Thank for your good project :+1:

**Comment by rcoup** [comment]
Hi @badele,

A [GeoPackage](https://en.wikipedia.org/wiki/GeoPackage) _is_ a SQLite database, with a couple of extra metadata tables. You can certainly _use_ a Kart repository/GeoPackage as a SQLite DB easily... though at the moment we use some of the GeoPackage metadata tables to identify dataset-related stuff which might make it slightly more complicated for writing.

It'd help for our planning/design if you can you explain your uses/use-case for SQLite?

**Reply by badele** [reply]
For my latest project https://github.com/badele/world-datas-analysis , i import frequently (mosly weekly) some datas from diverses datasets and i add the datas to `world-datas-analysis.db` sqlite database. 


If i understand the kart project., the idea would be to :

- **Week 1**
- i import and export data to `world-datas-analysis.db` + `kart commit`
- Others users do a `karl clone`

- **Week 2**
- i import and export data to `world-datas-analysis.db` + `kart commit`
- Others users do a `karl pull` (or equivalent command),only retrieve the diff with previous week

- **Week < # >**
- i import and export data to `world-datas-analysis.db` + `kart commit`
- Others users do a `karl pull` (or equivalent command), only retrieve the diff with previous week

Do you understand the idae ? And karl project can be respond this need ?

**Reply by rcoup** [reply]
Yes, I think Kart will work for your project. So there's a couple of options at the moment based on what you've described:

### A (preferred)

1. import `world-datas-analysis.db` once to a Kart repository using `kart import`. Kart will create a .gpkg working copy in the repository folder (it's a SQLite DB).
2. every week _update_ the tables in the working copy GPKG in your workflow 
3. `kart commit`
4. other users can do `kart clone`/`kart pull` to get updates

### B (slower, but will work)

1. import `world-datas-analysis.db` to a repository using `kart import`
2. every week your workflow creates a _new_ DB. Re-import this new DB into the existing repository using `kart import --replace-existing`
3. `kart commit`
4. other users can do `kart clone`/`kart pull` to get updates

One aspect to be aware of (for both approaches) is that while Kart supports tables without primary keys, everything will work _a lot better_ if you have primary keys present. `kart import` will pick them up if your DB schema defines them, or you can use `kart import --primary-key` to specify them manually.

**Reply by badele** [reply]
Really interesting, I really have to test this :)

Very thanks for your response :muscle:
